extensions-contrib/accurate-cardinality/src/main/java/org/apache/druid/query/aggregation/cardinality/accurate/types/StringAccurateCardinalityAggregatorColumnSelectorStrategy.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.cardinality.accurate.types;
+
+
+import org.apache.druid.segment.DimensionSelector;
+import org.apache.druid.segment.data.IndexedInts;
+
+public class StringAccurateCardinalityAggregatorColumnSelectorStrategy
+    implements AccurateCardinalityAggregatorColumnSelectorStrategy<DimensionSelector>
+{
+  @Override
+  public long getUniversalUniqueCode(DimensionSelector dimSelector)
```

Comment:
Is this truly universal I'm not familiar with the classes we're interacting with but I'm curious if distinct values from different segments could return colliding ints here.

org.apache.druid.query.aggregation.cardinality.types.StringCardinalityAggregatorColumnSelectorStrategy#hashValues looks up the actual value of the dimension before adding it to the cardinality aggregator.

Reply:
Maybe this code should be removed if it is invalid?